ExampleMod/Content/Items/Weapons/ExampleCustomUseStyleWeapon.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Microsoft.Xna.Framework;
+using System.IO;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items.Weapons
+{
+	// The classes in this file showcase making and using a custom use style. A use style controls the movement and hitbox of an item when used.
+	// This custom use style swings the sword up from below instead of the usual down swing of ItemUseStyleID.Swing.
+	// In addition, this custom use style can swing at any angle. 
+	// A separate example, ExampleCustomSwingSword, showcases an even more advanced custom swing using a held projectile instead of using custom use style code. It can be easier to implement advanced movements using a held projectile, but some may prefer the use style approach.
+
+	// ExampleCustomUseStyleGlobalItem contains the actual custom UseStyle logic. It is possible to put custom use style code directly in a ModItem, but we use a GlobalItem in this example because we use the same custom use style for the vanilla Cutlass weapon as well.
+	// ExampleCustomUseStyleItemSets stores extra data used by the custom item use style on a per-item type basis.
+	// ExampleCustomUseStylePlayer facilitates our custom use style.
+
+	public class ExampleCustomUseStyleWeapon : ModItem
+	{
+		public override void SetDefaults() {
+			// Here, we set the Item's useStyle to the custom use style value registered in ExampleCustomUseStyleGlobalItem.
+			// This will allow the custom logic in ExampleCustomUseStyleGlobalItem to run instead of the default logic for one of the existing useStyles.
+			// If instead of sharing custom use style code in a GlobalItem we wanted to make a custom use style for just this item, we could instead set Item.useStyle to -1 and implement the UseStyle, UseItemHitbox, and UseItemFrame methods in this class.
+			Item.useStyle = ExampleCustomUseStyleGlobalItem.ExampleCustomUseStyle;
+
+			Item.useAnimation = 20;
+			Item.useTime = 20;
+			Item.DamageType = DamageClass.Melee;
+			Item.damage = 20;
+			Item.width = 58;
+			Item.height = 58;
+			Item.knockBack = 2f;
+			Item.value = Item.sellPrice(gold: 1);
+			Item.rare = ItemRarityID.Green;
+			Item.autoReuse = true;
+			Item.UseSound = SoundID.Item1;
+
+			// This controls how far out the weapon should be held from the hand. This weapon uses 0 (so we don't actually need to set it) but the logic in ExampleCustomUseStyleGlobalItem works for other values as well. We can set it here or in the CreateIntSet method, both work.
+			//ExampleCustomUseStyleItemSets.HandOffsets[Type] = 0;
```

Comment:
Shouldn't this be in SetStaticDefaults instead?

Reply:
K.